core/src/main/java/jenkins/cli/StopBuildsCommand.java

```diff
@@ -0,0 +1,135 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.cli;
+
+import com.google.common.annotations.VisibleForTesting;
+import hudson.Extension;
+import hudson.cli.CLICommand;
+import hudson.model.Executor;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.model.Run;
+import jenkins.model.Jenkins;
+import org.kohsuke.args4j.Argument;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Extension
+public class StopBuildsCommand extends CLICommand {
+
+    private static final Logger LOGGER = Logger.getLogger(StopBuildsCommand.class.getName());
+
+    @VisibleForTesting
+    @Argument(usage = "Name of the job(s) to stop", required = true, multiValued = true)
+    /*private */ List<String> jobNames;
+
+    @Override
+    public String getShortDescription() {
+        return "Stop all running builds for job(s)";
+    }
+
+    @Override
+    protected int run() throws Exception {
+        Jenkins jenkins = Jenkins.get();
+        final HashSet<String> names = new HashSet<>();
+        names.addAll(jobNames);
+        final StringBuilder resultBuilder = new StringBuilder();
+        for (final String jobName : names) {
+            Item item = jenkins.getItemByFullName(jobName);
+            if (item instanceof Job) {
+                stopJobBuilds((Job) item, jobName, resultBuilder);
+            } else if (item != null) {
+                resultBuilder.append(String.format("Cannot abort runs for %s. Unsupported job type: %s\n", item, item.getClass()));
+            } else {
+                resultBuilder.append(String.format("Job with name %s not found.\n", jobName));
+            }
+        }
+
+        stdout.print(resultBuilder.toString());
+        return 0;
+    }
+
+    private void stopJobBuilds(final Job job,
+                               final String jobName,
+                               final StringBuilder resultBuilder) throws IOException, ServletException {
+        final Run lastBuild = job.getLastBuild();
+        final List<String> stoppedBuildsNames = new ArrayList<>();
+        if (lastBuild != null) {
+            stopBuild(lastBuild, jobName, stoppedBuildsNames);
```

Comment:
Since it was not immediately obvious to me—this _does_ verify that this build is in progress, inside `stopBuild`. (Since jobs may be concurrent-capable, the default for Pipeline in fact, the last build may be complete even while preceding builds are not.) So then the call from `checkAndStopPreviousBuilds` is redundantly checking `isBuilding`. Probably harmless if a bit confusing. I would suggest moving the `isBuilding` check to this call site since it is only for `getLastBuild` that we do not know whether or not we are dealing with an in-progress build. (Anyway a completed build would simply return null from `getExecutor`.)